File: DesignPatternsExercises/java/ObserverDesignPattern/Exercise/ObserverDesignPattern.java
public class ObserverDesignPattern {
	public static void main(String[] args) {
		BreakingNewsFeed breakingNewsFeed = new BreakingNewsFeed();

		new CNN(breakingNewsFeed);
		new FoxNews(breakingNewsFeed);

		breakingNewsFeed.setBreakingNews("UK votes to leave EU");
		System.out.println();

		new NBCNews(breakingNewsFeed);
		breakingNewsFeed.setBreakingNews("Dow Jones set new record");
	}
}

File: DesignPatternsExercises/ruby/observer_design_pattern/solution/observer_design_pattern.rb
require_relative 'breaking_news_feed'
require_relative 'cnn'
require_relative 'fox_news'
require_relative 'nbc_news'

breaking_news_feed = BreakingNewsFeed.new

CNN.new(breaking_news_feed)
FoxNews.new(breaking_news_feed)

breaking_news_feed.breaking_news = 'UK votes to leave EU'
puts ''

NBCNews.new(breaking_news_feed)
breaking_news_feed.breaking_news = 'Dow Jones set new record'
